app/src/main/java/com/keylesspalace/tusky/AccountActivity.java

```diff
@@ -417,10 +466,8 @@ public void onClick(View v) {
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
-            case R.id.action_back: {
-                Intent intent = new Intent(this, MainActivity.class);
-                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-                startActivity(intent);
+            case android.R.id.home: {
+                onBackPressed();
```

Comment:
In my experience, using this doesn't make the main view reload its contents from scratch. Though honestly, some kind of JSON caching is needed asap